graylog2-server/src/main/java/org/graylog/events/processor/EventProcessorExecutionJob.java

```diff
@@ -116,8 +115,12 @@ public JobTriggerUpdate execute(JobExecutionContext ctx) throws JobExecutionExce
             eventProcessorEngine.execute(config.eventDefinitionId(), parameters);
 
             // By using the processingWindowSize and the processingHopSize we can implement hopping and tumbling
-            // windows. (a tumbling window is simply a hopping window where windowSize and hopSize are the same)
-            DateTime nextTo = to.plus(config.processingHopSize());
+            // windows. (a tumbling window is simply a hopping window where windowSize and hopSize are the same).
+            // If the job uses cron scheduling, we need to instead calculate the nextTo field based on the current to
+            // field as it is possible to skip contiguous time ranges with cron scheduling.
+            DateTime nextTo = config.isCron() ?
+                    scheduleStrategies.nextTime(ctx.trigger(), to).orElse(to.plus(config.processingHopSize())) :
```

Comment:
When does `#nextTime(trigger, to)` return an empty Optional? The method's Javadoc says that the trigger shouldn't fire anymore if it returns an empty Optional.

Reply:
In this context, it would theoretically return an empty Optional only if the cron was specified to run for a given year/set of years and the event had run for the last time in the last year.

```java
    @Test
    public void emptyNextTimeCron() {
        final JobTriggerDto trigger = JobTriggerDto.builderWithClock(clock)
                .jobDefinitionId("abc-123")
                .jobDefinitionType("event-processor-execution-v1")
                .schedule(CronJobSchedule.builder()
                        .cronExpression("0 0 * ? * * 2024")
                        .build())
                .build();

        final DateTime nye2024 = DateTime.parse("2024-12-31T23:00:00.000Z");
        final Optional<DateTime> nextTime = strategies.nextTime(trigger, nye2024);

        assertThat(nextTime).isEmpty();
    }
```
In which case, the `nextTime` value would also be empty because the trigger's `lastNextTime` would have to be after the last `to` time. So the event would not run.